### DISCUSSION TITLE
About runtimeConfig.public

### DISCUSSION THREAD

**Discussion by pacmandoh** [nuxt/nuxt#24930] [Questions] [answered]
Will exposing the variables from .env to the ClientOnly component through runtimeConfig.public result in a leakage of the enviroment variables in .env?

thanks! ❤️

**Comment by dsvgl** [comment]
yep. It's called "public" for a reason ;)
https://nuxt.com/docs/guide/going-further/runtime-config

**Reply by pacmandoh** [reply]
Thanks for your response! Is it safe to pass environment variables to the ClientOnly component, or should I consider switching to SSR?

**Reply by TheAlexLichter** [reply]
@pacmandoh It depends. Everything that is part of `public` *will be accessible* in the frontend. For some services that might be necessary (think of stripe public key or google maps key). I've [explained more here](https://www.youtube.com/watch?v=2tKOZc3Z1dk).

**Reply by pacmandoh** [reply]
@manniL thx！ If my component needs to be client-side rendered, how can I securely pass environment variables?

**Reply by TheAlexLichter** [reply, marked as answer]
@pacmandoh As mentioned in the linked video and just before, you "cant". Everything in the frontend is human readable. You could call your API (assuming env vars containing keys for an API) through a nitro endpoint though, fully avoiding any exposure of the env variable (using it only in the API).

**Reply by pacmandoh** [reply]
Thx again for your response. I'll make sure to watch the video carefully. 💓
